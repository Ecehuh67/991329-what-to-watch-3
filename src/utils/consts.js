export const DEFAULT_GENRE = `All genres`;
export const DEFAULT_SHOWED_FILMS = 8;
export const DEFAULT_GENRES_AMOUNT = 9;
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  MY_LIST: `/mylist`,
  FILM: (id) => `/films/${id}`,
  PLAYER: (id) => `/player/${id}`,
  REVIEW: (id) => `/films/${id}/review`
}

export const EstimateValue = {
  3: "Bad",
  5: "Normal",
  8: "Good",
  10: "Very good",
  11: "Awesome"
};
