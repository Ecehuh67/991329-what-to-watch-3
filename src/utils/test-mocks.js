export const movies = [
  {
    title: `The Lord of the Rings`,
    image: `img/the-lord-of-the-rings.jpg`
  },
  {
    title: `The Ghostbusters`,
    image: `img/the-ghostbusters.jpg`
  },
  {
    title: `The Mortal Combat`,
    image: `img/the-mortal-combat.jpg`
  }
];
