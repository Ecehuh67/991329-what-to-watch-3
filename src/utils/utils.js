const MONTHS = {
  0: `January`,
  1: `February`,
  2: `March`,
  3: `April`,
  4: `May`,
  5: `June`,
  6: `Juily`,
  7: `Augest`,
  8: `September`,
  9: `October`,
  10: `November`,
  11: `December`
};

export const getRandomNumber = (max) => {
  return 1 + Math.floor(Math.random() * max);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const convertTime = (duration) => {
  const hours = Math.floor(duration / 60);
  const minutes = Math.round((duration / 60 - hours) * 60);

  return (
    `${hours}h ${minutes}m`
  );
};

export const convertTimeToProgressBar = (duration) => {
  const hours = Math.floor(duration / 3600);
  const minutes = Math.floor((duration - hours * 3600) / 60);
  const seconds = Math.round(duration - hours * 3600 - minutes * 60);
  return (
    `${hours}:${minutes}:${seconds < 10 ? `0${seconds}` : seconds}`
  );
};

export const getCommentDate = (date) => {
  const month = date.getMonth();
  const day = date.getDay();
  const year = date.getFullYear();

  return `${MONTHS[month]} ${day}, ${year}`;
};

export const getAttrDate = (date) => {
  const month = date.getMonth();
  const day = date.getDay();
  const year = date.getFullYear();

  return `${year}-${month < 10 ? `0${month}` : month}-${day < 10 ? `0${day}` : day}`;
};

export const getNewData = (newElement, arrayData) => {
  const newIndex = arrayData.findIndex((movie) => movie.id === newElement.id);
  const newData = arrayData.slice(0, newIndex).concat(arrayData.slice(newIndex + 1)).concat(newElement).sort((a, b) => a.id - b.id);

  return newData;
};

export const sortData = (newElement, arrayData) => {
  const newIndex = arrayData.findIndex((movie) => movie.id === newElement.id);
  const oldData = arrayData;

  if (newIndex === -1) {
    oldData.push(newElement);
  } else {
    oldData.splice(newIndex, 1);
  }

  return oldData;
};

export const getRank = (count) => {
  let rank;

  switch (true) {
    case count > 10:
      rank = `Awesome`;
      break;
    case count > 8 && count <= 10:
      rank = `Very good`;
      break;
    case count > 5 && count <= 8:
      rank = `Good`;
      break;
    case count > 3 && count <= 5:
      rank = `Normal`;
      break;
    case count > 0 && count <= 3:
      rank = `Bad`;
      break;
  }

  return rank;
};
