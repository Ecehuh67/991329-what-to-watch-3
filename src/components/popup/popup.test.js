import Popup from './popup';
import Tabs from '../../components/tabs/tabs';
import {movies} from '../../utils/test-mocks';
import {Route, BrowserRouter} from "react-router-dom";
import {AuthorizationStatus} from '../../utils/consts';

const BOOKMARKS_LIST = [`Overview`, `Details`, `Reviews`];
const DEFAULT_BOOKMARK = `Overview`;

const film = movies[1];

describe(`Render Popup correctly`, () => {
  it(`Render mock data the right way`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Route>
            <Popup
              film={film}
              films={movies}
              onDataChange={() => {}}
              addToFavorite={() => {}}
              loadComments={() => {}}
              authorizationStatus={AuthorizationStatus.NO_AUTH}
            >
              <Tabs
                state={
                  {
                    activeTab: DEFAULT_BOOKMARK,
                    tabsList: BOOKMARKS_LIST,
                    listener: () => {},
                    leaveComment: false,
                  }
                }
              />
            </Popup>
          </Route>
        </BrowserRouter>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
