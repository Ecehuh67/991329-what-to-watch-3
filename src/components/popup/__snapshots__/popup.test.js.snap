// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Popup correctly Render mock data the right way 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#000000",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Mortal Combat"
          src="https://player.vimeo.com/external"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Mortal Combat
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              thriller
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/19082"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <a
              className="btn btn--list movie-card__button"
              href="/login"
              onClick={[Function]}
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </a>
            <a
              className="btn movie-card__button"
              href="/films/19082/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Mortal Combat"
            height="327"
            src="https://player.vimeo.com/external/290130710.sd.mp4?s=7fa79288de952b505d7cc58e014712de048cac87&profile_id=164&oauth2_token_id=57447761"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/12"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <img
                alt="The ghoustbusters"
                height="175"
                src="https://player.vimeo.com/external/368010126.sd.mp4?s=ed2b36a42d2ac7b22000ed0f4ea187df7091423b&profile_id=139&oauth2"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              The ghoustbusters
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/19082"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <img
                alt="Mortal Combat"
                height="175"
                src="https://player.vimeo.com/external/368010126.sd.mp4?s=ed2b36a42d2ac7b22000ed0f4ea187df7091423b&profile_id=139&oauth2"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Mortal Combat
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1908"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <img
                alt="The time"
                height="175"
                src="https://player.vimeo.com/external/368010126.sd.mp4?s=ed2b36a42d2ac7b22000ed0f4ea187df7091423b&profile_id=139&oauth2"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              The time
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
